### PR TITLE
Add a evaluate keyword argument for Limit

### PR DESCRIPTION
**Pull request by amitsaha**
This will make it consistent with Derivative.

### CODE REVIEW

File: sympy/series/limits.py [hunk 1]
```diff
@@ -104,7 +106,10 @@ def __new__(cls, e, z, z0, dir="+"):
 
         obj = Expr.__new__(cls)
         obj._args = (e, z, z0, dir)
-        return obj
+        if not evaluate:
```
Comment (skirpichev) [inline sympy/series/limits.py]: you should respect global evaluate context.

Comment (amitsaha) [inline sympy/series/limits.py]: you mean nothing should be evaluated, not just the limit?

Comment (skirpichev) [inline sympy/series/limits.py]: I mean you should respect global evaluate context.  Please read documentation in core/evaluate.py, read other methods with evaluate kwarg.

**Comment by cbm755** [comment]
1.  Agree with @skirpichev's comment.  Here's an example from `Equality`:

````
    def __new__(cls, lhs, rhs=0, **options):
        lhs = _sympify(lhs)
        rhs = _sympify(rhs)

        evaluate = options.pop('evaluate', global_evaluate[0])
        ...
````

2.  Also, should `limit` (lowercase one) have evaluate kwarg too?  Then fully symmetric with how `diff(..., evaluate=False)` gives an unevaluated Derivative.

3.  Add a test?

**Comment by czgdp1807** [comment]
IMO, we don't need `evaluate` keyword as there is a already a function, `limit` which directly evaluates the limit. Probably we can close this.

**Comment by czgdp1807** [comment]
Closing this PR for now. Please re-open this if needed. Thanks for the contributions.

**Pull request closed without merging**